easier68k/core/opcodes/bcc.py

```diff
@@ -0,0 +1,737 @@
+
+        # offset to bits
+        self.offset = Bits(int=self.offset,
+                           length=(self.size.get_number_of_bytes() * 8)).int
+
+        # Note on Page 130 of the NXP 68K Manual
+        # 0x00 8 bit offset defaults to word size
+        if self.offset == 0:
+            self.size = OpSize.WORD
+
+    def conditional(self, simulator: M68K):
+        """
+        This is a method stub to be overridden in child classes
+        Checks the status register according to the specifics of 
+        whatever Bcc variant this is.
+        :param simulator: the M68K sim object that contains the CCR we are checking
+        :return bool: returns true or false 
+        """
+        raise NotImplementedError("Branch Operation did not override Parent's conditional stub")
+        
+    def assemble(self) -> bytearray:
+        """
+        Assembles opcode into Hex
+        :return: Hex for this operation
+        """
+        
+        # yes you actually need to subclass this
+        assert self.cond_code != None, 'No cond_code, assemble called on base clas of branch op?'
+
+        # hold size
+        extraSize = 2
+        
+        # clear for >8 bit offset
+        lastbyte = '\x00'
+        
+        # if offset is 8 bits than it is inside the opword
+        if self.size == OpSize.BYTE:
+            lastbyte = self.offset.to_bytes(length=1, byteorder='big', signed=True)
+            extraSize = 0
+
+        # if offset is 32 bits than the last chunk of opword is FF
+        if self.size == OpSize.LONG:
+            lastbyte = '\xFF'
+            extraSize = 4
+            
+        # first byte = 6+cond_code
+        firstbyte =  ord('\x60')
+        firstbyte |= ord(self.cond_code)
+        
+        opword = firstbyte << 8
+        opword |= ord(lastbyte)
+
+        # convert to bytearray
+        opword = bytearray(opword.to_bytes(2, 'big'))
+
+        # stop here if 8 bit offset
+        if self.size == OpSize.BYTE:
+            return opword
+
+        # add offset
+        opword.extend(self.offset.to_bytes(extraSize, "big"))
+
+        return opword
+
+    
+    def execute(self, simulator: M68K):
+        """
+        Commands the Simulator according to this opcode
+        :param simulator:
+        :return: nothing
+        """
+
+        # check conditional
+        if self.conditional(simulator):
+            self.offset += 2
+            simulator.increment_program_counter(self.offset)
+
+        pass
+
+    def __str__(self):
+        """
+        converts command to a string
+        :return: string rep of command
+        """
+
+        return "{s} ${d}".format(
+                s = COND_CODE_TO_OPCODE.get(self.cond_code),
+                d = hex(self.operand or
+                     (self.address + (self.size if self.size.get_number_of_bytes() > 1 else 0) + self.offset)))
+
+    @classmethod
+    def command_matches(cls, command: str) -> bool:
```

Comment:
We could also use the `@abstractmethod` decorator here. I think the order matters, I used both of these in #132